plugin/trino-google-sheets/src/main/java/io/trino/plugin/google/sheets/SheetsSplitManager.java

```diff
@@ -52,12 +53,19 @@ public ConnectorSplitSource getSplits(
             Constraint constraint)
     {
         SheetsTableHandle tableHandle = (SheetsTableHandle) connectorTableHandle;
-        SheetsTable table = sheetsClient.getTable(tableHandle.getTableName())
+        SheetsTable table = sheetsClient.getTable(tableHandle)
                 // this can happen if table is removed during a query
-                .orElseThrow(() -> new TableNotFoundException(tableHandle.toSchemaTableName()));
+                .orElseThrow(() -> {
+                    if (tableHandle.isNamedRelation()) {
+                        return new TableNotFoundException(tableHandle.asPlainTable().toSchemaTableName());
+                    }
+                    else {
+                        return new TableNotFoundException(new SchemaTableName("_generated", tableHandle.asSheetTable().getSheetExpression()));
```

Comment:
Was not sure what to do here since sheet tables do not have a schema or table name associated with them.

Reply:
Maybe add another subclass of `NotFoundException`... Reporting a synthesized name as the schema name, and a sheet id in place of a table name might be misleading.